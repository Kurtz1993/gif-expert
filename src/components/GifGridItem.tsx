import { FunctionComponent } from 'react';
import { GifInfo } from '../models/gif-info.model';

const GifGridItem: FunctionComponent<GifInfo> = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img className="card__img" src={url} alt={title} />
      <p className="card__title">{title}</p>
    </div>
  );
};

export default GifGridItem;
