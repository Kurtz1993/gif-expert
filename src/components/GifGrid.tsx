import { useFetchGifs } from '../hooks/use-fetch-gifs.hook';
import GifGridItem from './GifGridItem';

interface GifGridProps {
  category: string;
}

const GifGrid = ({ category }: GifGridProps) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {loading ? 'Loading...' : data.map(item => <GifGridItem key={item.id} {...item} />)}
      </div>
    </>
  );
};

export default GifGrid;
