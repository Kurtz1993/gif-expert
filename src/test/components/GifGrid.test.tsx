import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';
import GifGridItem from '../../components/GifGridItem';
import { useFetchGifs } from '../../hooks/use-fetch-gifs.hook';
import { GifInfo } from '../../models/gif-info.model';
jest.mock('../../hooks/use-fetch-gifs.hook');

describe('GifGridComponent', () => {
  const mockCategory = 'Overwatch';

  test('should render', () => {
    (useFetchGifs as jest.MockedFunction<typeof useFetchGifs>).mockReturnValue({
      images: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={mockCategory} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images are loaded via "useFetchGifs"', () => {
    const mockGifs: GifInfo[] = [
      {
        id: 'ABC',
        url: 'https://someething.com/algo.jpg',
        title: 'a title',
      },
    ];
    (useFetchGifs as jest.MockedFunction<typeof useFetchGifs>).mockReturnValue({
      images: mockGifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={mockCategory} />);

    expect(wrapper.find(GifGridItem).length).toBe(1);
  });
});
