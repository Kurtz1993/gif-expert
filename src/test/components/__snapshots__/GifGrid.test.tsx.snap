// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridComponent should render 1`] = `
<Fragment>
  <h3>
    Overwatch
  </h3>
  <div
    className="card-grid"
  >
    Loading...
  </div>
</Fragment>
`;
